Scenario/Editor/Models/ModelsUI.cs

```diff
@@ -171,25 +173,45 @@ public void OnGUI(Rect position)
 
         if (showPreviousButton && GUILayout.Button("Previous Page"))
         {
-            Models.GetModelsData(-1);
+            RunModelsDataOperation(-1);
         }
 
         if (showNextButton && GUILayout.Button("Next Page"))
         {
-            Models.GetModelsData(1);
+            RunModelsDataOperation(1);
         }
 
         GUILayout.EndHorizontal();
         GUILayout.EndArea();
     }
 
-    public async void UpdateTextures()
+    private async void RunModelsDataOperation(int direction)
+    {
+        await Models.GetModelsData(direction).ConfigureAwait(false);
+    }
+
+    public async Task UpdateTextures()
     {
         textures.Clear();
+        
         foreach (var item in pageList)
         {
-            Texture2D texture = await Models.LoadTexture(item.trainingImages[0].downloadUrl);
-            textures.Add((texture, item.name));
+            string downloadUrl = null;
+            
+            if (item.trainingImages != null && item.trainingImages.Count > 0)
+            {
+                downloadUrl = item.trainingImages[0].downloadUrl;
+            }
+            else if (item.thumbnail != null && !string.IsNullOrEmpty(item.thumbnail.url))
+            {
+                downloadUrl = item.thumbnail.url;
+            }
```

Comment:
I think we want it to be the other way around.
The thumbnail comes first, then the training image
See method in webapp
<img width="378" alt="image" src="https://github.com/scenario-labs/Scenario-Unity/assets/32990419/a6044891-f322-4489-ba19-1c3f213760ce">